test/EstateRegistry.js

```diff
@@ -0,0 +1,330 @@
+    const tokenId = await estate.estateTokenIds.call(estateId, index)
+    tokenId.toString().should.be.equal(value.toString())
+  }
+
+  beforeEach(setupRegistry)
+
+  describe('name', function() {
+    it('has a name', async function() {
+      const name = await estate.name()
+      name.should.be.equal(_name)
+    })
+  })
+
+  describe('symbol', function() {
+    it('has a symbol', async function() {
+      const symbol = await estate.symbol()
+      symbol.should.be.equal(_symbol)
+    })
+  })
+
+  describe('update metadata', async function() {
+    it('update works correctly', async function() {
+      const estateId = await createUserEstateWithToken1()
+      await estate.updateMetadata(estateId, newMsg, sentByUser)
+      await assertMetadata(estateId, newMsg)
+    })
+
+    it('unauthorized user can not update', async function() {
+      const estateId = await createUserEstateWithToken1()
+      await assertRevert(
+        estate.updateMetadata(estateId, newMsg, sentByAnotherUser)
+      )
+    })
+
+    it('unauthorized user can not set update operator', async function() {
+      const estateId = await createUserEstateWithToken1()
+      await assertRevert(
+        estate.setUpdateOperator(estateId, yetAnotherUser, sentByAnotherUser)
+      )
+    })
+
+    it('update operator can not transfer tokens out', async function() {
+      const estateId = await createUserEstateWithToken1()
+      await estate.setUpdateOperator(estateId, anotherUser, sentByUser)
+      await assertRevert(
+        estate.transferToken(estateId, 1, yetAnotherUser, sentByAnotherUser)
+      )
+    })
+  })
+
+  describe('transfer tokens', async function() {
+    it('owner can transfer tokens in', async function() {
+      const estateId = await createUserEstateWithToken1()
+      await land.assignMultipleParcels([0], [2], user, sentByCreator)
+      await transferIn(estateId, 2, user)
+      await assertEstateSize(estateId, 2)
+    })
+
+    it('random user can transfer tokens in', async function() {
+      const estateId = await createUserEstateWithToken1()
+      await land.assignMultipleParcels([0], [2], anotherUser, sentByCreator)
+      await transferIn(estateId, 2, anotherUser)
+      await assertEstateSize(estateId, 2)
+    })
+
+    it('random user can not transfer tokens out', async function() {
+      const estateId = await createUserEstateWithToken1()
+      await assertRevert(
+        estate.transferToken(estateId, 1, hacker, sentByAnotherUser)
+      )
+    })
+
+    it('random user can not transfer many tokens out', async function() {
+      const estateId = await createUserEstateWithToken1()
+      await assertRevert(
+        estate.transferManyTokens(estateId, [1], hacker, sentByAnotherUser)
+      )
+    })
+
+    it('owner can transfer tokens out', async function() {
+      const estateId = await createUserEstateWithToken1()
+      await estate.transferToken(estateId, 1, anotherUser, sentByUser)
```

Comment:
What do you think of also assert if `anotherUser` is the owner of the LandToken 1?. I know `safeTransferFrom` is a tested method, but It would be great to see if this line `dar.safeTransferFrom(this, destinatary, tokenId);` is working as expected

Reply:
Agreed!